### PR TITLE
Productionize Mercury

### PR DESCRIPTION
**Pull request by de846**

### CODE REVIEW

File: mercury/event_check.py [hunk 1]
```diff
@@ -15,6 +15,17 @@ def wrapper(self, request, *args, **kwargs):
...
+            valid_session = False
+        else:
+            # event is active and user does know the event code
+            valid_session = True
+
+        if request.session.get("event_code_active") is None:
+            # server hasn't determined if an event is active yet
+            valid_session = False
+
+        if not valid_session:
```
Comment (de846) [inline mercury/event_check.py]: Ignore, this is part of #267

File: mercury/tests/test_views.py [hunk 2]
```diff
@@ -99,6 +99,7 @@ def setUp(self):
         self.simulator_url = "mercury:simulator"
         self.can_url = "mercury:can-ui"
         self.stopwatch_url = "mercury:stopwatch"
+        self.client.get(reverse(self.login_url))
```
Comment (de846) [inline mercury/tests/test_views.py]: Ignore, this is part of #267

File: mercury/tests/test_views.py [hunk 3]
```diff
@@ -182,3 +183,12 @@ def test_bypass_login(self):
...
         self.assertEqual("index", response.url)
+
+
+class TestViewsWithoutCheckingEvent(TestCase):
+    def setUp(self):
+        self.index_url = "mercury:index"
+
+    def test_HomePageView_GET(self):
+        response = self.client.get(reverse(self.index_url))
+        self.assertEqual(302, response.status_code)
```
Comment (de846) [inline mercury/tests/test_views.py]: Ignore, this is part of #267

File: mysite/settings.py [hunk 4]
```diff
@@ -46,6 +48,9 @@
     "mercury.apps.MercuryConfig",
 ]
 SESSION_ENGINE = "django.contrib.sessions.backends.signed_cookies"
+SESSION_SAVE_EVERY_REQUEST = True
+SESSION_EXPIRE_AT_BROWSER_CLOSE = True
+SESSION_COOKIE_AGE = 3600
```
Comment (de846) [inline mysite/settings.py]: Ignore, this is part of #267

**Review by muneebafzal — APPROVED** [review]
LGTM, note: the check for coverage is failing rn

**Pull request merged**